source/includes/connection-snippets/scram-snippets/cpp-connection.cpp

```diff
@@ -0,0 +1,28 @@
+// begin scram connection
+#include <bsoncxx/builder/stream/document.hpp>
+#include <bsoncxx/json.hpp>
+#include <mongocxx/client.hpp>
+#include <mongocxx/uri.hpp>
+#include <mongocxx/instance.hpp>
+#include <bsoncxx/builder/basic/kvp.hpp>
+
+using bsoncxx::builder::basic::kvp;
+
+int main(int, char**) {
+
+mongocxx::instance inst{}; 
+auto client = mongocxx::client conn{
+   mongocxx::uri{
+      "mongodb+srv://<username>:<password>@<cluster-url>/test?retryWrites=true&w=majority"
+   }
+};
+
+mongocxx::database db = client["testDB"];
+mongocxx::collection collection = db["testCol"];
+bsoncxx::builder::basic::document basic_builder{};
+basic_builder.append(kvp("", ""));
```

Comment:
See https://github.com/mongodb/docs-ecosystem/pull/583/files#r352347017. This example should be updated to be consistent with the X509 example once you settle on the correct syntax.

I suppose a more general question is whether there's any way for the docs team to utilize the same template for each language example. The only bit that really needs to change between SCRAM and X509 is the construction of the MongoClient object, and most of that is actually just the connection string. If there's some way to extract that bit out and utilize a common program template, that may make it easier to ensure that the programs themselves are correct and also facilitate the addition of future auth examples.

Ideally, I think drivers would own these language examples and run them as part of their test suite as we do for CRUD and transactions (@kay-kim is familiar with that), but I assume there's a reason that method was not utilized here.